pkg/util/util.go

```diff
@@ -143,3 +185,44 @@ func SanitizeName(name string) string {
 	}
 	return name
 }
+
+func GetObjectKey(obj interface{}) (string, error) {
+	if unknown, ok := obj.(cache.DeletedFinalStateUnknown); ok && unknown.Obj != nil {
+		obj = unknown.Obj
+	}
+	objKey, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get key from object")
+		return "", err
+	}
+	return objKey, nil
+}
+
+// Patches a given PVC with changes from newPVC. If addResourceVersionCheck is true
+// then a version check is added to the patch to ensure that we are not patching
+// old(and possibly outdated) PVC objects.
+func PatchClaim(kubeClient kubernetes.Interface, oldPVC, newPVC *v1.PersistentVolumeClaim, addResourceVersionCheck bool) (*v1.PersistentVolumeClaim, error) {
+	patchBytes, err := GetPVCPatchData(oldPVC, newPVC, addResourceVersionCheck)
+	if err != nil {
+		return oldPVC, fmt.Errorf("can't patch status of PVC %s as generate path data failed: %v", klog.KObj(oldPVC), err)
+	}
+	updatedClaim, updateErr := kubeClient.CoreV1().PersistentVolumeClaims(oldPVC.Namespace).
+		Patch(context.TODO(), oldPVC.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{}, "status")
```

Comment:
This method does not update the cache anymore, will this impact the resizer a lot? @gnufied

Reply:
hmm, that is bit of a problem, because I don't want this PR to accidentally change the flow in resizer controller for now. Can we update the calling sites to still update the cache stores same as before? 

So something like:

```
pvc, err := ctrl.patchClaim(pvc, newPVC, true /* addResourceVersionCheck */)
if err != nil {
   ...
}

	err = ctrl.claims.Update(updatedClaim)
	if err != nil {
		return oldPVC, fmt.Errorf("error updating PVC %s in local cache: %v", klog.KObj(newPVC), err)
	}
```

Reply:
We can do this too if you think it is safe.